pandas/core/reshape/merge.py

```diff
@@ -938,6 +938,8 @@ def _create_join_index(
         -------
         join_index
         """
+        if how == "left" and self.how == "asof":
```

Comment:
This seems very specific. Is this the deepest level at which this should be handled?

Reply:
I think so, yes. The error actually occurs there. Everything before is handled as expected in ``_get_join_info``. The only issue is the call of  ``_create_join_index`` with the wrong parametrization.

We could implement the function  ``_get_join_info`` for merge_asof too, but the function does more than just this part so we would duplicate a lot probably?

Reply:
hmm, it seems that myabe asof should be in hit in _get_join_info (there is a case that is excluding). I agree this is not a great place to put this. rather it would pay to trace the asof paths and see if there is a more general problem. the conditions in _get_join_info are slightly convoluted .